framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -311,8 +422,168 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		return false;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	private async _createCCU() {}
+	private async _calculateInboxUpdate(sideChainID: Buffer): Promise<InboxUpdate> {
+		const serialzedCCMs = this._chainConnectorState.crossChainMessages.map(ccm =>
+			codec.encode(ccmSchema, ccm),
+		);
+		const ccmHashes = serialzedCCMs.map(ccm => utils.hash(ccm));
+
+		const channel = await this._mainchainAPIClient.invoke<ChannelData>(
+			'interoperability_getChannel',
+			{ sideChainID },
+		);
+
+		let newInboxRoot;
+		let newInboxAppendPath = channel.inbox.appendPath;
+		let newInboxSize = channel.inbox.size;
+
+		for (const ccm of ccmHashes) {
+			const { appendPath, size, root } = regularMerkleTree.calculateMerkleRoot({
+				value: ccm,
+				appendPath: newInboxAppendPath,
+				size: newInboxSize,
+			});
+			newInboxAppendPath = appendPath;
+			newInboxSize = size;
+			newInboxRoot = root;
+		}
+
+		const db = new InMemoryDatabase();
+
+		const smt = new SparseMerkleTree({
+			db,
+			rootHash: channel.inbox.root,
+		});
+
+		const outboxKey = this._rawStateStoreKey(sideChainID);
+
+		await smt.update(outboxKey, newInboxRoot as Buffer);
+
+		const proofSMT = await smt.generateSingleProof(outboxKey);
+
+		const outboxRootWitness: OutboxRootWitness = {
+			bitmap: Buffer.from(proofSMT.binaryBitmap),
+			siblingHashes: proofSMT.siblingHashes,
+		};
+
+		const inboxUpdate: InboxUpdate = {
```

Comment:
We changed the params schema: https://github.com/LiskHQ/lips/blob/update-interoperability-LIPs/proposals/lip-0053.md#parameters

in particular, the inboxUpdate now does not contain the partnerChainOutboxSize

Reply:
@has5aan Update the params schema as described in the LIP update https://github.com/LiskHQ/lips/blob/update-interoperability-LIPs/proposals/lip-0053.md#parameters